DIPs/DIP1xxx-SSS.md

```diff
@@ -0,0 +1,184 @@
+the data in the destination location for the object, as that has no pointers pointing at it. Such modifications will require a
+cast.
+
+Whether it is safe to modify external data pointed to by pointers stored in the struct (such as the case for an intrusive linked
+list), or the data in the source address of the move, heavily depends on the specifics of the implementation. The user
+documentation for `opPostMove` MUST explain what is guaranteed to be safe and what is not.
+
+Implementors may also choose to not define const/immutable versions of `opPostMove`. This will results in a compile-time error should
+the compiler try to move such a struct.
+
+The documentation for `opPostMove` MUST also emphasize that while manipulating the memory at the `opPostMove` source location is allowed,
+the memory will be effectively freed with no destruction immediately after the function's return. Implementors SHOULD be
+encouraged to define the argument to `opPostMove` to be `const ref` to gain some compiler protection against accidental manipulation.
+This does not harm the implementor's access to the data, as she already has a copy at the destination location.
+
+### Code emitted by the compiler on move
+
+When moving a struct, the compiler should call `__move_post_blt` on the struct, giving it both new and old instances.
+
+### `opPostMove` Decoration Considerations
+
+Ideally, `opPostMove` should be `@nogc`, `nothrow` and either `@safe` or `@trusted`. If that doesn't happen, trying to compile
+code that moves a struct from context that is `nothrow`, `@nogc` or `@safe` would result in a compilation error.
+
+We could force these by decorating `__move_post_blt` itself as `nothrow @nogc @safe`, thus not allowing `opPostMove` to be defined
+any other way (we cannot otherwise force `opPostMove` to be defined any specific way, as it is being defined by the user). I chose
+not to do so because the user might opt not to use, e.g., `@safe` anywhere in her program. It would, therefor, not make sense
+to force her to use it in `opPostMove`. Due to attribute inference on template functions, if all member `opPostMove`s are, e.g., `@nogc`,
+D will automatically define `__move_post_blt` as `@nogc`.
+
+We do force `nothrow` on `opPostMove`, because throwing would mean a change in the program flow from a place that does not seem to
+run code, which might prove too confusing.
+
+## Performance Considerations
+
+Structs that do not define `opPostMove`, and that none of their members define `opPostMove`, will have their `__move_post_blt`
+implementation be just a function calling a bunch of empty functions recursively. Hopefully, the compiler will be able to inline
+this series of calls into oblivion, meaning the run time cost of this feature for structs that do not use it will be zero.
+
+If the compiler implementers fear that inlining will not nullify these calls where not applicable, they MAY manually eliminate
```

Comment:
"If the compiler implementers" -> If compiler implementers fear

"nullify these calls where not applicable" -> elide these calls where applicable